src/lib.rs

```diff
@@ -0,0 +1,736 @@
+        })
+    }
+
+    fn to_zenoh_key_string(uri: &UUri) -> Result<String, UStatus> {
+        let micro_uuri = MicroUriSerializer::serialize(uri).map_err(|_| {
+            UStatus::fail_with_code(
+                UCode::INVALID_ARGUMENT,
+                "Unable to serialize into micro format",
+            )
+        })?;
+        let micro_zenoh_key: String = micro_uuri
+            .iter()
+            .fold(String::new(), |s, c| s + &format!("{c:02x}"));
+        Ok(micro_zenoh_key)
+    }
+
+    #[allow(clippy::match_same_arms)]
+    fn map_zenoh_priority(upriority: UPriority) -> Priority {
+        match upriority {
+            UPriority::UPRIORITY_CS0 => Priority::Background,
+            UPriority::UPRIORITY_CS1 => Priority::DataLow,
+            UPriority::UPRIORITY_CS2 => Priority::Data,
+            UPriority::UPRIORITY_CS3 => Priority::DataHigh,
+            UPriority::UPRIORITY_CS4 => Priority::InteractiveLow,
+            UPriority::UPRIORITY_CS5 => Priority::InteractiveHigh,
+            UPriority::UPRIORITY_CS6 => Priority::RealTime,
+            // If uProtocol prioritiy isn't specified, use CS1(DataLow) by default.
+            // https://github.com/eclipse-uprotocol/uprotocol-spec/blob/main/basics/qos.adoc
+            UPriority::UPRIORITY_UNSPECIFIED => Priority::DataLow,
+        }
+    }
+
+    fn to_upayload_format(encoding: &Encoding) -> Option<UPayloadFormat> {
+        let Ok(value) = encoding.suffix().parse::<i32>() else {
+            return None;
+        };
+        UPayloadFormat::from_i32(value)
+    }
+
+    // TODO: Transformation between uAttributes and attachment
+    //       Transform all members first and then exclude some we don't need
+    fn uattributes_to_attachment(uattributes: &UAttributes) -> anyhow::Result<AttachmentBuilder> {
+        let mut attachment = AttachmentBuilder::new();
+        attachment.insert("id", &uattributes.id.write_to_bytes()?);
+        attachment.insert(
+            "type_",
+            &uattributes
+                .type_
+                .enum_value()
+                .map_err(|_| UStatus::fail_with_code(UCode::INTERNAL, "Unable to parse type"))?
+                .to_type_string(),
+        );
+        attachment.insert("source", &uattributes.source.write_to_bytes()?);
+        attachment.insert("sink", &uattributes.sink.write_to_bytes()?);
+        // TODO: Check whether request & response need priority or not
+        attachment.insert("priority", &uattributes.priority.value().to_string());
+        if let Some(ttl) = uattributes.ttl {
+            attachment.insert("ttl", &ttl.to_string());
+        }
+        if let Some(plevel) = uattributes.permission_level {
+            attachment.insert("permission_level", &plevel.to_string());
+        }
+        if let Some(commstatus) = uattributes.commstatus {
+            attachment.insert("commstatus", &commstatus.to_string());
+        }
+        attachment.insert("reqid", &uattributes.reqid.write_to_bytes()?);
```

Comment:
only response messages contain a *reqid*, so I guess this should be done conditionally only